kernel/src/transaction/create_table.rs

```diff
@@ -0,0 +1,991 @@
+        // Process column mapping (assigns IDs and physical names if enabled)
+        let column_mapping = Self::process_column_mapping(schema, &extracted.cleaned_properties)?;
+
+        // Get the column mapping mode for use in data layout processing
+        let column_mapping_mode =
+            get_column_mapping_mode_from_properties(&column_mapping.properties)?;
+
+        // Update properties with column mapping results (includes maxColumnId if CM enabled)
+        extracted.cleaned_properties = column_mapping.properties;
+
+        // Add column mapping features if enabled
+        for feature in column_mapping.additional_reader_features {
+            if !extracted.reader_features.contains(&feature) {
+                extracted.reader_features.push(feature);
+            }
+        }
+        for feature in column_mapping.additional_writer_features {
+            if !extracted.writer_features.contains(&feature) {
+                extracted.writer_features.push(feature);
+            }
+        }
+
+        Ok((column_mapping.schema, column_mapping_mode))
+    }
+
+    /// Ensures no table exists at the given path by checking for the _delta_log directory.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if a Delta table already exists at the path.
+    fn ensure_table_does_not_exist(
+        table_url: &Url,
+        path: &str,
+        engine: &dyn Engine,
+    ) -> DeltaResult<()> {
+        let delta_log_url = table_url.join("_delta_log/")?;
+        let storage = engine.storage_handler();
+
+        // Try to list the _delta_log directory - if it exists and has files, table exists
+        match storage.list_from(&delta_log_url) {
+            Ok(mut files) => {
+                if files.next().is_some() {
+                    return Err(Error::generic(format!(
+                        "Table already exists at path: {}",
+                        path
+                    )));
+                }
+            }
+            Err(_) => {
+                // Directory doesn't exist, which is what we want for a new table
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Validates clustering columns against the schema and constraints.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if:
+    /// - More than 4 clustering columns are specified
+    /// - A clustering column doesn't exist in the schema
+    fn validate_clustering_columns(columns: &[ColumnName], schema: &SchemaRef) -> DeltaResult<()> {
+        use crate::schema::DataType;
+
+        // Check maximum clustering columns
+        if columns.len() > MAX_CLUSTERING_COLUMNS {
+            return Err(Error::generic(format!(
+                "Cannot specify more than {} clustering columns. Found {}.",
+                MAX_CLUSTERING_COLUMNS,
+                columns.len()
+            )));
+        }
+
+        // Validate each column exists in the schema by traversing the path
+        for col in columns {
```

Comment:
Hm -- do we have no faster/easier way to do this? Seems like a good standalone utility that might already exist?